src/pipeline/src/dispatcher.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use snafu::OptionExt;
+use yaml_rust::Yaml;
+
+use crate::etl::error::{Error, Result};
+use crate::etl_error::{
+    FieldRequiredForDispatcherSnafu, TablePartRequiredForDispatcherRuleSnafu,
+    ValueRequiredForDispatcherRuleSnafu,
+};
+use crate::Value;
+
+const FIELD: &str = "field";
+const TABLE_PARTIAL: &str = "table_part";
+const PIPELINE: &str = "pipeline";
+const VALUE: &str = "value";
+const RULES: &str = "rules";
+
+/// The dispatcher configuration.
+///
+/// Dispatcher in a pipeline allows user to call another pipeline and specify
+/// table name based on field matching.
+///
+/// ```yaml
+/// dispatcher:
+///   field: type
+///   rules:
+///     - value: http
+///       pipeline: http_pipeline
+///       table_part: http_log
+///     - value: db
+///       pipeline: db_pipeline
+///       table_part: db_log
+///
+///
+/// ```
+///
+/// If none of the rules match the value, this pipeline will continue to process
+/// current log entry
+#[derive(Debug, PartialEq)]
+pub(crate) struct Dispatcher {
+    pub field: String,
+    pub rules: Vec<Rule>,
+}
+
+/// The rule definition for dispatcher
+///
+/// - `value`: for pattern matching
+/// - `pipeline`: the pipeline to call, if it's unspecified, we use default
+///   `greptime_identity`
+/// - `table_part`: the table name segment that we use to construct full table
```

Comment:
what's "part" (partial) means

Reply:
Let me add more description for this in next patch.
This partial is to construct a full table name when the write request is dispatched to another pipeline. In that situation, we cannot use original table name from input, but to construct a new table using this pattern and input table name. 

Perhaps table_suffix is a name easier to understand. Will update this in next patch.